### PR TITLE
Revert options bags

### PR DESCRIPTION
**Pull request by jaschrep-msft**

### CODE REVIEW

File: sdk/storage/Azure.Storage.Blobs/src/BlobBaseClient.cs [hunk 1]
```diff
@@ -1032,6 +1032,7 @@ private void SetNameFieldsIfNull()
         ///     </item>
         /// </list>
         /// </remarks>
+        [EditorBrowsable(EditorBrowsableState.Never)]
```
Comment (seanmcc-msft) [inline sdk/storage/Azure.Storage.Blobs/src/BlobBaseClient.cs:1035]: Why are we hiding `DownloadStreaming()`?

File: sdk/storage/Azure.Storage.Blobs/src/BlobBaseClient.cs [hunk 2]
```diff
@@ -1114,6 +1115,7 @@ private void SetNameFieldsIfNull()
...
```
Comment (seanmcc-msft) [inline sdk/storage/Azure.Storage.Blobs/src/BlobBaseClient.cs:1118]: Same as above.

**Review by seanmcc-msft — APPROVED** [review]
Approved pending CI

File: sdk/storage/Azure.Storage.Blobs/src/BlobBaseClient.cs (hunk 1 above)
Comment (jaschrep-msft) [reply sdk/storage/Azure.Storage.Blobs/src/BlobBaseClient.cs:1035]: These should have been hidden already. This is the only method we did not remove the options bag from, as it got another feature elsewhere. This corrects that mistake, leaving us with a sync/async pair of options bag overloads visible and this sync/async pair hidden. The current state follows the pattern we want to follow.

**Comment by jaschrep-msft** [comment]
For reference, subject to editing:

- Blobs
  - Append block should be reverted. 
  - Stage block should be reverted.
  - Upload pages should be reverted.
  - **Upload remains (never had an older hashing api)**
  - Download Streaming/Content should keep options to support IProgress; add byte[] to the options bag to support both features
  - **DownloadTo remains (never had an older hashing api)**
- DataLake
  - Append should be reverted
  - **Upload remains (never had an older hashing api)**
  - Read should be reverted
  - **ReadTo remains (never had an older hashing api)**
- Shares
  - Upload range should be reverted
  - Upload should be reverted
  - Download should be reverted
  - DownloadTo doesn't exist
- **Open Read/Write remains across packages**

File: sdk/storage/Azure.Storage.Blobs/src/BlobBaseClient.cs (hunk 1 above)
Comment (jaschrep-msft) [reply sdk/storage/Azure.Storage.Blobs/src/BlobBaseClient.cs:1039]: Resolved offline that this is the correct state.

**Pull request merged**